dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/QueryLogSharedState.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.time.ZonedDateTime;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class QueryLogSharedState {
+  public static final ConcurrentMap<QueryLogEntry, ZonedDateTime> queryLogEntries =
```

Comment:
The concurrent collection suggests that this class is designed for use from multiple threads. If this is the case, please document how this is done - is it fully thread-safe and what rules need to be followed by developers using it. If any guarantees provided by the class aren't obvious then it's also good to document them.